native/common/jp_gc.cpp

```diff
@@ -237,10 +245,24 @@ void JPGarbageCollection::onEnd()
 		Py_ssize_t pred = current + 2 * (current - last);
 		last = current;
 		if ((Py_ssize_t) pred > (Py_ssize_t) limit)
+		{
 			run_gc = 2;
+            limit = high_water + (high_water>>3) + 8 * (current - last);
```

Comment:
can you please explain this heuristic? What does it aim for?

Reply:
The current logic is that if Python memory was rising above the high water (or predicted to) then it would trigger then java gc.   When the python gc is complete we were supposed to reset the high water so that we won't run the java gc again for a while.  

In Windows, it appears that rather than stop the world Python is still running. So what is happening is that in the example sent we triggered the java gc over and over and over.   To prevent that I have to raise the limit above the waves so that it won't trigger unless garbage is actually piling up.  Here is a graph of before and after.

![image](https://github.com/user-attachments/assets/0d1ca76a-42cd-4e6b-a512-514a181d367e)